### PR TITLE
ShootAndSeedSameRegion - Schema & Feature Flag

### PR DESCRIPTION
**Pull request by ralikio**
**Description**

Changes proposed in this pull request:

- new parameter `shootAndSeedSameRegion` added to the KEB schema,
- feature flag to disable rendering of the parameter.

**Related issue(s)**

### CODE REVIEW

File: internal/broker/plans.go [hunk 1]
```diff
@@ -427,11 +427,15 @@ func empty() *map[string]interface{} {
 	return &empty
 }
 
-func createSchemaWithProperties(properties ProvisioningProperties, additionalParams, update bool, requiered []string) *map[string]interface{} {
+func createSchemaWithProperties(properties ProvisioningProperties, additionalParams, update bool, requiered []string, shootAndSeedSameRegion bool, shootAndSeedFeatureFlag bool) *map[string]interface{} {
 	if additionalParams {
 		properties.IncludeAdditional()
 	}
 
+	if shootAndSeedFeatureFlag && additionalParams && shootAndSeedSameRegion {
```
Comment (jaroslaw-pieszka) [inline internal/broker/plans.go:435]: Why you check additionalParams here?

File: internal/broker/plans.go [hunk 2]
```diff
@@ -427,11 +427,15 @@ func empty() *map[string]interface{} {
 	return &empty
 }
 
-func createSchemaWithProperties(properties ProvisioningProperties, additionalParams, update bool, requiered []string) *map[string]interface{} {
+func createSchemaWithProperties(properties ProvisioningProperties, additionalParams, update bool, requiered []string, shootAndSeedSameRegion bool, shootAndSeedFeatureFlag bool) *map[string]interface{} {
```
[the remark this answers is not in the document]
Comment (ralikio) [reply internal/broker/plans.go]: Corrected.

File: internal/broker/plans.go (hunk 1 above)
Comment (ralikio) [reply internal/broker/plans.go:435]: I treat this shootAndSeedSameRegion as additionalParam. Beside that, it is used to hide in endpoints and tests to hide additional parameter section depending on the context.

Comment (jaroslaw-pieszka) [reply internal/broker/plans.go:435]: So someone calling `createSchemaWithProperties` should set both `additionalParams` and `shootAndSameRegion` to true if he intends to enforce the same region.

**Review by jaroslaw-pieszka — APPROVED** [review]

**Pull request merged**